File: backend/models/Order.js
const mongoose = require('mongoose');

const orderSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: [true, 'User is required']
  },
  items: [{
    product: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Product',
      required: true
    },
    productName: {
      type: String,
      required: true
    },
    quantity: {
      type: Number,
      required: true,
      min: 1
    },
    price: {
      type: Number,
      required: true,
      min: 0
    }
  }],
  total: {
    type: Number,
    required: [true, 'Total is required'],
    min: [0, 'Total cannot be negative']
  },
  paymentMethod: {
    type: String,
    enum: ['razorpay', 'cod', 'upi'],
    required: [true, 'Payment method is required']
  },
  paymentStatus: {
    type: String,
    enum: ['pending', 'completed', 'failed', 'refunded'],
    default: 'pending'
  },
  status: {
    type: String,
    enum: ['pending', 'processing', 'shipped', 'delivered', 'cancelled'],
    default: 'pending'
  },
  shippingAddress: {
    street: { type: String, required: true },
    city: { type: String, required: true },
    state: { type: String, required: true },
    zipCode: { type: String, required: true },
    country: { type: String, default: 'India' },
    phone: { type: String, required: true }
  },
  razorpayOrderId: {
    type: String
  },
  paymentId: {
    type: String
  },
  shiprocketOrderId: {
    type: String
  },
  shipmentId: {
    type: String
  },
  trackingUrl: {
    type: String
  },
  timeline: [{
    status: String,
    timestamp: { type: Date, default: Date.now },
    note: String
  }]
}, {
  timestamps: true
});

// Add timeline entry when status changes
orderSchema.pre('save', function(next) {
  if (this.isModified('status')) {
    this.timeline.push({
      status: this.status,
      timestamp: new Date(),
      note: `Order status changed to ${this.status}`
    });
  }
  next();
});

module.exports = mongoose.model('Order', orderSchema);

File: lib/models/order.dart
class Order {
  final String id;
  final String userName;
  final double total;
  final String status;
  final String paymentStatus;
  final String? shiprocketOrderId;
  final String? trackingUrl;
  final String? shipmentId;
  final DateTime createdAt;

  Order({
    required this.id,
    required this.userName,
    required this.total,
    required this.status,
    required this.paymentStatus,
    this.shiprocketOrderId,
    this.trackingUrl,
    this.shipmentId,
    required this.createdAt,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['_id'] ?? '',
      userName: json['user'] is Map ? (json['user']['name'] ?? 'Guest') : 'Guest',
      total: (json['amount'] as num?)?.toDouble() ?? 0.0,
      status: json['orderStatus'] ?? 'pending',
      paymentStatus: json['paymentStatus'] ?? 'pending',
      shiprocketOrderId: json['shiprocketOrderId'],
      trackingUrl: json['trackingUrl'],
      shipmentId: json['shipmentId'],
      createdAt: DateTime.parse(json['createdAt'] ?? DateTime.now().toIso8601String()),
    );
  }
}
